crates/exex/exex/src/backfill/stream.rs

```diff
@@ -72,15 +76,18 @@ where
     /// Spawns a new task calling the [`BackfillTaskIterator::next`] method and pushes it to the
     /// [`BackfillTasks`] queue.
     fn push_task(&mut self, mut job: BackfillTaskIterator<T>) {
-        self.tasks.push_back(tokio::task::spawn_blocking(move || (job.next(), job)));
+        self.tasks.push_back(tokio::task::spawn_blocking(move || BackfillTaskOutput {
+            result: job.next(),
+            job,
+        }));
     }
 
     /// Polls the next task in the [`BackfillTasks`] queue until it returns a non-empty result.
     fn poll_next_task(&mut self, cx: &mut Context<'_>) -> Poll<Option<BackfillJobResult<T>>> {
         while let Some(res) = ready!(self.tasks.poll_next_unpin(cx)) {
```

Comment:
this returns None if empty, right?

Reply:
yep, if there are no more backfill jobs left in the queue, it returns `None` and it means the backfill has finished for the whole range